modules/timer/timer.cpp

```diff
@@ -0,0 +1,327 @@
+static void  copyInitialToTemporalValuesOfTheActiveMode ();
+static void  assignValuesForANewCicle                   ();
+
+//=====[Implementations of public functions]===================================
+void checkValuesOfCurrentModeAndReassignIfCorresponds() {
+    if (currentMode == MODE_EXERCISE) {
+        if (
+            exerciseModeTemporalValues[3]   == 0  && 
+            exerciseModeTemporalValues[2]   == 0  && 
+            exerciseModeTemporalValues[1]   == 0  && 
+            exerciseModeTemporalValues[0]   == 0
+        ) {
+            currentMode = MODE_REST;
+            return;
+        } else {
+            return;
+        }
+    }
+
+    if (currentMode == MODE_REST) {
+        if (
+            restModeTemporalValues[3]   == 0  && 
+            restModeTemporalValues[2]   == 0  && 
+            restModeTemporalValues[1]   == 0  && 
+            restModeTemporalValues[0]   == 0
+        ) {
+            currentMode = MODE_REPETITIONS;
+            return;
+        } else {
+            return;
+        }
+    }
+
+    if (currentMode == MODE_REPETITIONS) {
+        if (
+            repetitionModeTemporalValues[3]   == 0  && 
+            repetitionModeTemporalValues[2]   == 0  && 
+            repetitionModeTemporalValues[1]   == 0  && 
+            repetitionModeTemporalValues[0]   == 0
+        ) {
+            setCountdownState(false);
+            informThatCountdownIsFinished();
+            currentMode = MODE_EXERCISE;
+            return;
+        } else {
+            assignValuesForANewCicle();
+            return;
+        }
+    }
+    return;
+}
+
+int * readInitialValueOfTheModeActive() {
+    switch (currentMode){
+        case MODE_EXERCISE:     return exerciseModeInitialValues;
+        case MODE_REST:         return restModeInitialValues;
+        case MODE_REPETITIONS:  return repetitionModeInitialValues;
+        default:                return errorValues;
+    }
+}
+
+int * readTemporalValueOfTheModeActive() {
+    switch (currentMode){
+        case MODE_EXERCISE:     return exerciseModeTemporalValues;
+        case MODE_REST:         return restModeTemporalValues;
+        case MODE_REPETITIONS:  return repetitionModeTemporalValues;
+        default:                return errorValues;
+    }
+}
+
+modes_t readModeActive() {
+    return currentMode;
+}
+
+void updateMode() {
+    switch (currentMode){
+        case MODE_EXERCISE:     
+            currentMode = MODE_REST;
+            break;
+        case MODE_REST:         
+            currentMode = MODE_REPETITIONS;
+            break;
+        case MODE_REPETITIONS:  
+            currentMode = MODE_EXERCISE;
+            break;
+        default:                
+            currentMode = MODE_EXERCISE;
+            break;
+    }
+
+    copyInitialToTemporalValuesOfTheActiveMode(); // I notice that is necessary to make this call at least for the case when the first button pressed is BUTTON_SELECT_MODE 
```

Comment:
Idem comentario previo sobre poner comentarios tan largos en dos líneas.

Reply:
Ok, lo cambio.

Reply:
Irresolví la conversación porque me olvidé de incluir este cambio. En el commit ['add comments to led and timer modules'](https://github.com/ferauzalez/ARMBookCurso2024_TP1/pull/3/commits/9a832068cde91f1d3f34c751d1d5ac6ccc3f79cc) agrego el cambio. No sé como es el procedimiento para notificarte específicamente esto. Por favor marcá como resuelto nuevamente cuando lo veas.